rules/framework.bzl

```diff
@@ -270,9 +270,17 @@ def _get_framework_files(ctx, deps):
                 destination = paths.join(framework_dir, "PrivateHeaders", hdr.basename)
                 private_header_out.append(destination)
 
+        has_header = False
+        if CcInfo in dep:
+            for hdr in dep[CcInfo].compilation_context.direct_headers:
```

Comment:
This block is pretty similar below - consider we have a way to extract all hdrs from providers, then looping that 
```
for hdr in _get_dep_hdrs(dep):
```

Reply:
I was unsure about the `.modulemap`... I _think_ that got cleaned up in `rules_swift` as well no?

Reply:
Under `rules_ios` we now propagate that via hdrs, given that, I wonder if we also need this from `CcInfo` if it's not picked up anywhere else. Maybe related to why the CI is failing?

Reply:
```
bazel-out/ios-x86_64-min10.0-applebin_ios-ios_x86_64-dbg-ST-a7ed3b5187e2/bin/tests/ios/in-tree-vendor-prebuilt-deps/dynamic/app/FW/FW.framework/Headers/FW-umbrella.h:20:1: error: umbrella header for module 'FW' does not include header 'FW_Private.h' [-Werror,-Wincomplete-umbrella]
tests/ios/in-tree-vendor-prebuilt-deps/dynamic/app/App/main.m:1:9: fatal error: could not build module 'FW'
@import FW;
 ~~~~~~~^~
```

Reply:
good catch, that was exactly it.